File: typescript-client-sdk/src/sdk/models/operations/getbatchcrmemail.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetBatchCrmEmailQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=ids" })
  ids?: string;
}


export class GetBatchCrmEmailResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=emails", elemType: shared.Email })
  emails?: shared.Email[];

  @SpeakeasyMetadata({ data: "json, name=invalidIds" })
  invalidIds?: string[];
}


export class GetBatchCrmEmailRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetBatchCrmEmailQueryParams;
}


export class GetBatchCrmEmailResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetBatchCrmEmailResponseBody;
}

File: go-client-sdk/pkg/models/operations/getbatchcrmemail.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetBatchCrmEmailQueryParams struct {
	AccessToken string  `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool   `queryParam:"style=form,explode=true,name=allFields"`
	Ids         *string `queryParam:"style=form,explode=true,name=ids"`
}

type GetBatchCrmEmailResponseBody struct {
	Emails     []shared.Email `json:"emails,omitempty"`
	InvalidIds []string       `json:"invalidIds,omitempty"`
}

type GetBatchCrmEmailRequest struct {
	QueryParams GetBatchCrmEmailQueryParams
}

type GetBatchCrmEmailResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetBatchCrmEmailResponseBody
}
